src/hotspot/share/gc/shenandoah/shenandoahVerifier.cpp

```diff
@@ -347,14 +347,17 @@ class ShenandoahVerifyOopClosure : public BasicOopIterateClosure {
 // a subset (e.g. the young generation or old generation) of the total heap.
 class ShenandoahCalculateRegionStatsClosure : public ShenandoahHeapRegionClosure {
 private:
-  size_t _used, _committed, _garbage, _regions;
+  size_t _used, _committed, _garbage, _regions, _humongous_waste;
 public:
-  ShenandoahCalculateRegionStatsClosure() : _used(0), _committed(0), _garbage(0), _regions(0) {};
+  ShenandoahCalculateRegionStatsClosure() : _used(0), _committed(0), _garbage(0), _regions(0), _humongous_waste(0) {};
 
   void heap_region_do(ShenandoahHeapRegion* r) {
     _used += r->used();
     _garbage += r->garbage();
     _committed += r->is_committed() ? ShenandoahHeapRegion::region_size_bytes() : 0;
+    if (r->is_humongous()) {
+      _humongous_waste += r->free();
```

Comment:
Wouldn't this be the case only when `_used > 0` (i.e. the HR is in use and isn't free? Otherwise, is there an implicit assumption of the regions over which this closure is working that wants to appear as an assertion check here?)

Reply:
I'm not sure I follow. This closure visits every region. We don't want to count the `free` portions of regular/in-use regions as humongous waste.

Reply:
> a fortiori
>: with greater reason or more convincing force —used in drawing a conclusion that is inferred to be even more certain than another

TIL (Today I Learned)